braindecode/datautil/windowers.py

```diff
@@ -0,0 +1,200 @@
+"""Get epochs from mne.Raw
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import mne
+import pandas as pd
+
+
+class Windower(object):
```

Comment:
Updating my comments from above now that I've seen the latest version:

If I understand correctly, we have a new set of Windower classes which (1) use a new internal function (_supercrop_starts) instead of mne.events_from_annotations to identify events, (2) use 'crop'/'supercrop' language instead of 'windows' in the variable names and docstring and (3) set the default behaviour to drop_samples=False (instead of requiring the user to set this explicitly).

1. Before I forget, I think docstrings should start with a one-line short description on the same line as the first """: see https://www.python.org/dev/peps/pep-0008/#documentation-strings
2. Why did we chose to use a new internal function instead of the existing MNE functions? Personally, I think the closer we stay to the MNE way of doing things, the simpler it will be to maintain the package. Also, people already familiar with MNE don't have to learn a new way of extracting windows.
3.  Do you think we should keep the crop/supercrop-specific language? My worry is that since this class is central to the package, and these concepts are relatively niche, they might make the understanding of the library more difficult. As far as I understand, supercrop could just be replaced by window here as there is no subdivision into crops happening at this level.
4. I understand that you guys' use case is to extract a window that covers the last "missed" samples in each event. I still believe that this default behaviour can be problematic in some cases (as the opposite behaviour can be problematic in others). I liked our previous solution to this which was to force the user to provide drop_samples - what do you think?

Reply:
1. I wasn't aware of this. Thanks for pointing this out. I will pay more attention to this in the future.
2. As @robintibor pointed out, the supercrop/window logic creation did not translate well to MNE and added even more complexity to the problem. Just think about tmin, tmax, chunk_duration (which changes its behavior), etc.. There is actually no need to use MNE at this point. To me, the new _supercrop_starts function is clearer than everything we had before.
3. @robintibor please decide on the language and we can then stick to it.
4. I am sorry, I forgot about this. I think, it will be best to not set a default and force the user to make a decision. Therefore, he will then also be responsible for the consequences.

Reply:
Thanks for the answers! :)

2. Maybe we could discuss with @agramfort whether there's a modification to MNE's API that would make this simple here? I agree that the new function is clearer to read, but ultimately I think offloading whatever can be done by MNE to MNE would help in maintaining the codebase.